uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/manager-leave-requests.html';
     }],
-    controllerAs: 'requests',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'Contact', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'OptionGroup', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, Contact, AbsencePeriod, AbsenceType, LeaveRequest, OptionGroup) {
     $log.debug('Component: manager-leave-requests');
+
+    var vm = Object.create(this),
+      allStatus = {
```

Comment:
We need the Array to show the statuses in the UI using `ng-repeat`, so we need an `Array` . Changed the implementation though and removed the variable.